base_layer/core/tests/helpers/block_builders.rs

```diff
@@ -337,3 +338,31 @@ pub fn generate_block_with_coinbase(
     }
     result
 }
+
+// This will return a new block for use in testing, it will not add the block to the database or vec of blocks
+pub fn calculate_new_block(
```

Comment:
This fn is basically 1 line different from `generate_new_block`. No need to replicate all that functionality when you could extract the relevant function.